onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc

```diff
@@ -203,26 +203,40 @@ bool ResizeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers
   }
 
   {  // scales and sizes (if present) must be initializers.
-    if (input_defs.size() < 3) {
-      LOGS(logger, VERBOSE) << "Input scales or sizes of Resize must be known";
+    if (input_defs.size() < 3 && node.SinceVersion() == 11) {
+      LOGS(logger, VERBOSE) << "Input scales of Resize must be known for opset 11";
       return false;
     }
 
-    // scales
-    if (input_defs.size() == 3 && !Contains(initializers, input_defs[2]->Name())) {
-      LOGS(logger, VERBOSE) << "Input scales of Resize must be known";
-      return false;
+    // scales (scales maybe empty tensor)
+    bool has_scales = false;
+    if (input_defs.size() == 3) {
+      has_scales = !input_defs[2]->Name().empty();
+      if ((has_scales && !Contains(initializers, input_defs[2]->Name())) ||
+          (!has_scales && node.SinceVersion() == 11)) {
+        LOGS(logger, VERBOSE) << "Input scales of Resize must be known";
+        return false;
+      }
+    }
+
+    // sizes (sizes maybe empty tensor)
```

Comment:
Consider flattening this a bit so there's less `if` nesting for the brain to follow:

```
    // sizes (sizes may be an empty tensor)
    bool has_sizes = input_defs.size() >= 3 && !input_defs[3]->Name().empty();
    if (has_sizes && !Contains(initializers, input_defs[3]->Name())) {
      LOGS(logger, VERBOSE) << "Input sizes of Resize must be known";
      return false;
    }
```

`maybe` --> `may be` (verb rather than adverb)